### PR TITLE
Fix Create team button not working in header

### PR DESCRIPTION
**Pull request by shige**
## Summary
- Fix Create team button that was not responding to clicks
- Move `canCreateFreeTeam` logic to server-side for security

## Problem
PR #2328 introduced a regression where the "Create team" button in the header and navigation rail was rendered as a plain `<span>` element instead of using `TeamCreationForm`, making it non-functional.

Additionally, the previous implementation in `navigation-rail-footer-menu.tsx` computed `canCreateFreeTeam` on the client side, which could be bypassed.

## Solution
1. Created `canCreateFreeTeam` function in `services/teams/plan-features/` to centralize the logic
2. Compute `canCreateFreeTeam` in the server-side data loader
3. Pass the value to client components via the existing data loading pattern
4. Use `TeamCreationForm` component to wrap the "Create team" text

## Test plan
https://github.com/giselles-ai/giselle/pull/2348#issuecomment-3605537258

___

### **PR Type**
Bug fix, Enhancement

___

### **Description**
- Move `canCreateFreeTeam` logic to server-side for security

- Fix non-functional Create team button in header and navigation

- Add comprehensive test coverage for free team creation rules

- Centralize team creation permission checks across components

___

### Diagram Walkthrough

```mermaid
flowchart LR
  A["data-loader.ts<br/>Server-side"] -->|"canCreateFreeTeam"| B["UserDataForNavigationRail<br/>Type"]
  B -->|"Pass to components"| C["header.tsx"]
  B -->|"Pass to components"| D["navigation-rail-footer-menu.tsx"]
  B -->|"Pass to components"| E["team-selection-compact.tsx"]
  C -->|"Render"| F["TeamCreationForm<br/>with validation"]
  D -->|"Render"| F
  E -->|"Render"| F
```

<details> <summary><h3> File Walkthrough</h3></summary>

<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>free-team-creation.ts</strong><dd><code>Create server-side free team creation validation</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

apps/studio.giselles.ai/services/teams/plan-features/free-team-creation.ts

<ul><li>New function <code>canCreateFreeTeam</code> that validates user eligibility<br> <li> Checks if user is internal (route06.co.jp domain)<br> <li> Checks if user already has existing free team<br> <li> Returns boolean indicating creation permission</ul>

</details>

  </td>
  <td><a href="https://github.com/giselles-ai/giselle/pull/2348/files#diff-3d214065c27be306b65b33d80f34cb4ea029b8667fb12405ac0f5d55601514e7">+20/-0</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>data-loader.ts</strong><dd><code>Compute server-side free team creation permission</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

apps/studio.giselles.ai/app/stage/data-loader.ts

<ul><li>Import <code>canCreateFreeTeam</code> function from plan-features<br> <li> Call <code>canCreateFreeTeam</code> with user email and team plans<br> <li> Add <code>canCreateFreeTeam</code> to returned navigation data object<br> <li> Set <code>canCreateFreeTeam</code> to false in error fallback case</ul>

</details>

  </td>
  <td><a href="https://github.com/giselles-ai/giselle/pull/2348/files#diff-e24a35f1efe9deaf040a8799b804e42b2f045d4cf639d648187440d1d542c264">+7/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>types.ts</strong><dd><code>Add canCreateFreeTeam to navigation data type</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

apps/studio.giselles.ai/app/stage/ui/navigation-rail/types.ts

<ul><li>Add <code>canCreateFreeTeam: boolean</code> property to <code>UserDataForNavigationRail</code> <br>interface<br> <li> Enables passing server-computed permission to client components</ul>

</details>

  </td>
  <td><a href="https://github.com/giselles-ai/giselle/pull/2348/files#diff-b85747f98e392de0ee0131956ad6b0cb7410024fe4eb62211dba6121c724a78f">+1/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>
</table></td></tr><tr><td><strong>Tests</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>free-team-creation.test.ts</strong><dd><code>Add comprehensive test coverage for validation</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

apps/studio.giselles.ai/services/teams/plan-features/free-team-creation.test.ts

<ul><li>Add 8 test cases covering all validation scenarios<br> <li> Test internal user restrictions<br> <li> Test existing free team restrictions<br> <li> Test edge cases with null/undefined email<br> <li> Test multiple team scenarios</ul>

</details>

  </td>
  <td><a href="https://github.com/giselles-ai/giselle/pull/2348/files#diff-d0c51731044dbc49be780c74327d5e2ca0ad522b443886477aa4b9d68e5a7c8d">+35/-0</a>&nbsp; &nbsp; </td>

</tr>
</table></td></tr><tr><td><strong>Bug fix</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>header.tsx</strong><dd><code>Fix Create team button in header dropdown</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

apps/studio.giselles.ai/app/stage/ui/header.tsx

<ul><li>Import <code>TeamCreationForm</code> component<br> <li> Replace non-functional <code><span>Create team</span></code> with <code>TeamCreationForm</code><br> <li> Pass <code>canCreateFreeTeam</code> and <code>proPlanPrice</code> props to component<br> <li> Apply consistent styling to Create team text</ul>

</details>

  </td>
  <td><a href="https://github.com/giselles-ai/giselle/pull/2348/files#diff-bbe6b15e28aefe1228682521aee118037f0837397521a7e8186b8b6295e88d45">+11/-1</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>navigation-rail-footer-menu.tsx</strong><dd><code>Remove client-side validation, use server value</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

apps/studio.giselles.ai/app/stage/ui/navigation-rail/navigation-rail-footer-menu.tsx

<ul><li>Remove client-side <code>canCreateFreeTeam</code> computation logic<br> <li> Remove inline email validation and team checking code<br> <li> Use server-computed <code>user.canCreateFreeTeam</code> instead<br> <li> Simplify component by delegating permission logic to server</ul>

</details>

  </td>
  <td><a href="https://github.com/giselles-ai/giselle/pull/2348/files#diff-a0516a4b8e699a7fb52a4036a4126ccb5f9e836d13a5b55002b9d82b9e221e90">+1/-9</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>team-selection-compact.tsx</strong><dd><code>Fix Create team button in compact selection</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

apps/studio.giselles.ai/app/stage/ui/navigation-rail/team-selection-compact.tsx

<ul><li>Import <code>TeamCreationForm</code> component<br> <li> Replace non-functional <code><span>Create team</span></code> with <code>TeamCreationForm</code><br> <li> Pass <code>canCreateFreeTeam</code> and <code>proPlanPrice</code> props to component<br> <li> Apply consistent styling to Create team text</ul>

</details>

  </td>
  <td><a href="https://github.com/giselles-ai/giselle/pull/2348/files#diff-4428e874bfbb158c179ad36f713626f6e2e1756bf0d719b79c696c6a4b1ab21a">+11/-1</a>&nbsp; &nbsp; </td>

</tr>
</table></td></tr></tr></tbody></table>

</details>

___

## Summary by CodeRabbit

* **New Features**
  * Team creation now respects eligibility: free-team creation option is shown or blocked based on account eligibility.
  * Team creation UI consolidated into a dedicated creation form and now displays plan pricing info during creation flows.

* **Behavior Changes**
  * Ineligible users are prevented from creating free teams (attempts now show an error/are blocked).

* **Tests**
  * Added comprehensive tests covering free-team eligibility scenarios.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

---

> [!NOTE]
> Restores functional Create team UI and centralizes free-team eligibility on the server with enforcement and tests.
> 
> - **Backend**:
>   - **Eligibility logic**: Add `canCreateFreeTeam` in `services/teams/plan-features/free-team-creation.ts` and use it in `data-loader.ts` and `actions/create-team.ts` (blocks ineligible free team creation).
>   - **Types**: Extend `UserDataForNavigationRail` with `canCreateFreeTeam`.
> - **Frontend**:
>   - **Team creation UI**: Replace static `"Create team"` spans with `TeamCreationForm` in `app/stage/ui/header.tsx` and `navigation-rail/team-selection-compact.tsx`.
>   - **Footer menu**: Remove client-side eligibility checks; use `user.canCreateFreeTeam` in `navigation-rail-footer-menu.tsx`.
> - **Tests**:
>   - Add `free-team-creation.test.ts` covering internal users and existing free-team scenarios.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 7c7ae0e52794d75b935185c33844c8e6e2633802. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: apps/studio.giselles.ai/services/teams/plan-features/free-team-creation.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+	existingTeamPlans: TeamPlan[],
+): boolean {
+	const isInternalUser = email != null && isEmailFromRoute06(email);
+	if (isInternalUser) {
+		return false;
+	}
+
+	const hasExistingFreeTeam = existingTeamPlans.some((plan) => plan === "free");
+	return !hasExistingFreeTeam;
+}
```
[the remark this answers is not in the document]
Comment (shige) [reply apps/studio.giselles.ai/services/teams/plan-features/free-team-creation.ts:20]: fixed 7c7ae0e52794d75b935185c33844c8e6e2633802

**Comment by shige** [comment]
Thank you! 🚀
